crates/solidity/outputs/cargo/crate/src/backend/semantic/mod.rs

```diff
@@ -0,0 +1,138 @@
+use std::collections::BTreeMap;
+use std::rc::Rc;
+
+use semver::Version;
+
+use crate::backend::binder::Binder;
+pub use crate::backend::ir::ir2_flat_contracts as output_ir;
+use crate::backend::passes;
+use crate::backend::types::TypeRegistry;
+use crate::compilation::File;
+use crate::cst::{Cursor, NonterminalNode};
+use crate::parser::ParseError;
+
+pub struct SemanticBuilder {
+    language_version: Version,
+    files: Vec<Rc<File>>,
+}
+
+impl SemanticBuilder {
+    pub fn create(language_version: Version) -> Self {
+        Self {
+            language_version,
+            files: Vec::new(),
+        }
+    }
+
+    pub fn add_file(&mut self, file: Rc<File>) {
+        self.files.push(file);
+    }
+
+    pub fn build(self) -> SemanticAnalysis {
+        SemanticAnalysis::create(self.language_version, self.files)
+    }
+}
```

Comment:
Is there any particular need for a Builder here? I'm not against it, but it seems the only place we use it we could just call `SemanticAnalysis::create(self.language_version, self.files)`